Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -740,20 +746,23 @@ bool WebFrame::isFrameSet() const
 
 bool WebFrame::isMainFrame() const
 {
-    return m_coreFrame && m_coreFrame->isMainFrame();
+    RefPtr localCoreFrame = coreFrame();
+    return localCoreFrame && m_coreFrame->isMainFrame();
 }
 
 bool WebFrame::isRootFrame() const
 {
-    return m_coreFrame && m_coreFrame->isRootFrame();
+    RefPtr localCoreFrame = coreFrame();
```

Comment:
You keep using `local` in the names of local variables but this is confusing, especially for frames where we already have the local vs remote distinction (due to site isolation) and which has a different meaning.

Normally, we'd just use:
```
RefPtr coreFrame = this->coreFrame();
```
or
```
RefPtr frame = coreFrame();
```